numpyro/distributions/directional.py

```diff
@@ -0,0 +1,62 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+import math
+
+import jax.numpy as jnp
+from jax import lax, dtypes
+
+from numpyro.distributions import constraints
+from numpyro.distributions.distribution import Distribution
+from numpyro.distributions.util import validate_sample
+from numpyro.util import copy_docs_from
+from numpyro.distributions.util import promote_shapes, von_mises_centered
+
+
+@copy_docs_from(Distribution)
+class VonMises(Distribution):
+    arg_constraints = {'loc': constraints.interval(-math.pi, math.pi), 'concentration': constraints.positive}
```

Comment:
I prefer to leave it as before: `constraints.real` (i.e. if users provide `loc = 3pi/2`, the mean of this distribution will be `-pi / 2`) to be more flexible but it totally depends on you. If you think this constraint is good, could you add docs mention that `loc` is expected to be in (-pi, pi) interval?

Reply:
I'll update it to the use the real.